File: android2_lesson04/src/java/java/com/example/ama/android2_lesson04/ServiceTestApp.java
package com.example.ama.android2_lesson04;

import android.app.Application;

import com.example.ama.android2_lesson04.data.Data;

public class ServiceTestApp extends Application {
    private static ServiceTestApp instance;
    private static Data data;

    public static ServiceTestApp getInstance() {
        return instance;
    }

    public static Data getData() {
        return data;
    }

    @Override
    public void onCreate() {
        super.onCreate();
        instance = this;
        data = new Data();
    }
}

File: android2_lesson04/src/kotlin/java/com/example/ama/android2_lesson04/ServiceTestApp.kt
package com.example.ama.android2_lesson04

import android.app.Application
import com.example.ama.android2_lesson04.data.Data

class ServiceTestApp : Application() {

    companion object {
        lateinit var instance: ServiceTestApp
        lateinit var data: Data
    }

    override fun onCreate() {
        super.onCreate()
        instance = this
        data = Data()
    }
}